backend/server.js

```diff
@@ -1,67 +1,107 @@
-const path = require('path')
-const express = require('express')
-const app = express()
-const port = process.env.PORT || 8080
-const mysql = require('mysql')
-const bodyParser = require('body-parser')
-require('./run-database-migrations')
+const path = require("path");
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 8080;
+const mysql = require("mysql");
+const bodyParser = require("body-parser");
+require("./run-database-migrations");
 
 const pool = mysql.createPool({
   connectionLimit: 20,
-  host     : process.env.RDS_HOSTNAME || 'localhost',
-  user     : process.env.RDS_USERNAME || 'root',
-  password : process.env.RDS_PASSWORD || 'password',
-  database : process.env.RDS_DB_NAME || 'local_db',
-  port     : process.env.RDS_PORT || '3306',
+  host: process.env.RDS_HOSTNAME || "localhost",
+  user: process.env.RDS_USERNAME || "root",
+  password: process.env.RDS_PASSWORD || "password",
+  database: process.env.RDS_DB_NAME || "local_db",
+  port: process.env.RDS_PORT || "3306"
 });
 
-app.set('views', path.join(__dirname, '/views'));
-app.set('view engine', 'ejs')
-app.use('/static', express.static(path.join(__dirname, '../static')))
-app.use(bodyParser.urlencoded({ extended: false }))
+app.set("views", path.join(__dirname, "/views"));
+app.set("view engine", "ejs");
+app.use("/static", express.static(path.join(__dirname, "../static")));
+app.use(bodyParser.urlencoded({ extended: false }));
 app.use(bodyParser.json());
 
-app.post('/api/users', (req, res, next) => {
+app.post("/api/duplicate-check/", (req, res, next) => {
+  /*Tried to add this route in the routes directory but dont know how to carry to mysql query connection over to that... Is this the right way? */
   pool.getConnection((err, connection) => {
     if (err) {
-      return databaseError(req, res, err)
+      return databaseError(req, res, err);
+    }
+    let d = new Date(req.body.birthday);
+    let year = d.getUTCFullYear();
+    let month = d.getUTCMonth() + 1;
+    let day = d.getUTCDate();
+    let inserts = [
+      "person",
+      req.body.firstname,
+      req.body.lastname,
+      year,
+      month,
+      day
+    ];
+    let qryString =
+      "SELECT personid,firstname,lastname,date_format(dob,'%m/%d/%Y')as birthDate,gender FROM ?? WHERE ";
+    qryString +=
+      "((firstname = ? OR lastname = ?) AND (year(dob) = ? OR month(dob) = ? OR day(dob) = ?))";
```

Comment:
This code will report someone named Joel Denning born on May 5, 1980 a duplicate of someone named Joel Denning who was born on December 5, 1990. Which to me doesn't make much sense. My first thought is we should just check the whole date, instead of the month day and year separately? Something like this:

```js
mysql.format('SELECT ... AND dob = ?', [req.body.birthday])
```